R/remotes.R

```diff
@@ -145,6 +188,19 @@ renv_remotes_parse_finalize <- function(parsed) {
 
 }
 
+renv_remotes_is_git <- function(parsed) {
+  return(
+    "git" %in% c(
```

Comment:
Can you clarify why this is necessary?

Reply:
I'll leave a comment, but essentially I wanted to be able to parse a git url that wasn't given explicitly with a type, eg:

git@github.com:user/repo.git

The hints are the git@, and the .git

Reply:
I'm at a computer so I can explain further- I want the following to work: `git@github.com:user/repo.git`

Reply:
May I ask - Are you asking me to add a comment, or were you asking me right now what the change is for?

Reply:
Maybe it's not necessary to infer type without the `git::` prefix - Also - given that `https://` git links don't work correctly without the `git` prefix, and it's probably be a fair bit more work to resolve that, perhaps this inconsistency implies the extra complexity is not worth it.

Interested in your thoughts @kevinushey

Reply:
@kevinushey would you like me to remove this type inference?

Reply:
I think so. I think any of the inference needs to happen up-front when `renv_remotes_parse_git()` is called (assuming it detects what appears to be a compatible `git` reference). Once that type has been inferred, it can be saved and reused.

In other words, if the code reaches here:

https://github.com/rstudio/renv/pull/762/files#diff-c30ff3320c29175c880b55adb3cb4229d4b0840e9a6f9df162ec8760216a063dR174

then I think you could just "force" the setting of the type to be "git", since the git parser successfully detected it. Does that make sense?

Reply:
We could also take the view that this parser only considers entries that end with `.git`, since it would be unlikely that any other entry type would use that suffix. That might help simplify some pieces and remove potential ambiguities with the other parsers.

Reply:
> We could also take the view that this parser only considers entries that end with `.git`, since it would be unlikely that any other entry type would use that suffix.

To clarify - do you mean that this parser should reject an entry that does not end with `.git`, for example:

- Works:
  - `git::git@github.com:user/repo.git`
- Fails:
  - `git::git@github.com:user/repo`

If so, I disagree with your intepretation, as `git@github.com:user/repo` is a valid route on most git servers.

Or do you mean - the parser should only use the explicit `git::` prefix, or `.git` extension to infer `git` type?

Unfortunately, without more extensive changes, the `.git` extension wouldn't force `git` type inference reliably, as in:

- This will fail unless I change more stuff
  - `https://github.com/user/repo.git` 
  - `https` is handled specially, earlier.

> That might help simplify some pieces and remove potential ambiguities with the other parsers.